go/vt/log/log.go

```diff
@@ -78,5 +81,32 @@ var (
 // calls this function, or call this function directly before parsing
 // command-line arguments.
 func RegisterFlags(fs *pflag.FlagSet) {
-	fs.Uint64Var(&glog.MaxSize, "log_rotate_max_size", glog.MaxSize, "size in bytes at which logs are rotated (glog.MaxSize)")
+	flagVal := logRotateMaxSize{
+		val: "1887436800", // glog.MaxSize value (which is not concurrency safe)
+	}
+	fs.Var(&flagVal, "log_rotate_max_size", "size in bytes at which logs are rotated (glog.MaxSize)")
```

Comment:
Simply passing the literal as the default — rather than the `glog.MaxSize` var — is likely enough. If you prefer I can do that instead:
```
fs.Uint64Var(&glog.MaxSize, "log_rotate_max_size", 1887436800, "size in bytes at which logs are rotated (glog.MaxSize)")
```

Reply:
does this work?

```go
flagVal := logRotateMaxSize{val: fmt.Sprintf("%d", glog.MaxSize)}
fs.Var(&flagVal, ...)
```

that way if a glog version were to ever happen to change the value (unlikely, but nonzero), we'd get the change for free

Reply:
It works, yes. But was still a little racy, even when using an atomic load there.